akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -345,10 +348,32 @@ object StreamLayout {
       if (s != shape) throw new UnsupportedOperationException("cannot replace the shape of the EmptyModule")
       else this
 
-    override def compose(that: Module): Module = that
-
-    override def compose[A, B, C](that: Module, f: (A, B) ⇒ C): Module =
-      throw new UnsupportedOperationException("It is invalid to combine materialized value with EmptyModule")
+    override def compose(that: Module): Module = compose(that, scaladsl.Keep.left)
```

Comment:
The default was equvalent to `Keep.right`, which is wrong, because:

1. GraphDSL.create() with no imports starts with EmptyModule, so final materialized value should be `NotUsed`
2. Now with `.add(g)` we might add a graph with a materialized value, but since we don't want to alter the final type, so `Keep.left` is the right behavior.

The `Keep.right` version is still needed though in the case of single-graph import, since then `.add(g)` called from the `create()` implementation should replace the materialized value with that of `g`.